src/GraphQL/Types/Scalars/EnumerationGraphType.cs

```diff
@@ -62,39 +62,48 @@ public void AddValue(EnumValueDefinition value)
         public EnumValues Values { get; }
 
         /// <inheritdoc/>
-        public override object Serialize(object value)
+        public override object ParseLiteral(IValue value) => value switch
         {
-            var valueString = value.ToString(); //TODO: find only by value?
-            var foundByName = Values.FindByName(valueString);
-            if (foundByName != null)
-            {
-                return foundByName.Name;
-            }
+            EnumValue enumValue => Values.FindByName(enumValue.Name)?.Value ?? ThrowLiteralConversionError(value),
+            NullValue _ => null,
+            _ => ThrowLiteralConversionError(value)
+        };
```

Comment:
We even had a TODO note in here to remove searching by name for Serialize.  Now fixed.

Reply:
Yes, I suspected for a long time that something was wrong.